src/rust/engine/fs/store/src/remote.rs

```diff
@@ -81,6 +81,14 @@ async fn choose_provider(options: RemoteOptions) -> Result<Arc<dyn ByteStoreProv
       "byte-store".to_owned(),
       options,
     )?))
+  } else if let Some(url) = address.strip_prefix("github-actions-cache+") {
+    // TODO: this is relying on python validating that it was set as
+    // `github-actions-cache+https://...`
```

Comment:
Is this actually a TODO, or just a note? If the former, explaining what should happen instead would be good.

Reply:
It's a confusing semi-TODO (sorry!): as referenced in the PR description, I don't like the `<provider>+<url>` approach in practice (partly, but not entirely, because this validation is awkward), and so I've gone with this crappy approach here as a good-enough solution until I refactor it away.

That is, the TODO is a poorly-phrased attempt to say "improve this verification once the inputs are simpler"... but it'd be just better as a note.